src/databricks/labs/ucx/workspace_access/generic.py

```diff
@@ -189,6 +198,25 @@ def _safe_get_permissions(self, object_type: str, object_id: str) -> iam.ObjectP
             else:
                 raise RetryableError() from e
 
+    def _safe_update_permissions(
+        self, object_type: str, object_id: str, acl: list[iam.AccessControlRequest]
+    ) -> iam.ObjectPermissions | None:
+        try:
+            return self._ws.permissions.update(object_type, object_id, access_control_list=acl)
+        except DatabricksError as e:
+            if e.error_code in [
+                "BAD_REQUEST",
+                "INVALID_PARAMETER_VALUE",
+                "UNAUTHORIZED",
+                "PERMISSION_DENIED",
+                "FEATURE_DISABLED",
+                "RESOURCE_DOES_NOT_EXIST",
+            ]:
+                logger.warning(f"Could not update permissions for {object_type} {object_id} due to {e.error_code}")
+                return None
+            else:
+                raise RetryableError() from e
```

Comment:
what error codes are we retrying then?... please add a comment, otherwise it's not clear